tools/benchmark/tests-code.js

```diff
@@ -0,0 +1,606 @@
+var b = require('benchmark-pyret');
+require('jasmine-node');
+
+jasmine.getEnv().addReporter(new jasmine.ConsoleReporter(console.log));
+
+var validProgram = '1';
+var invalidProgram = '1 + true';
+var nonParsableProgram = '...';
+// var longProgram = '';
+// for(var i = 0; i < 1000; i++){
+//   longProgram += '1 + ';
+// }
+// longProgram += '1';
+
+
+///SKIP ALL TESTS INVOLVING longProgram FOR NOW
+// problem with async waiting.
+describe('checkResult', function(){
+  it('returns true given a SuccessResult', function(){
+    expect(b.test.testCheckResult(true)).toBe(true);
+  });
+
+  it('returns false given a FailureResult', function(){
+    expect(b.test.testCheckResult(false)).toBe(false);
+  });
+});
+
+describe('ensureSuccess', function() {
+  it('passes on a valid program', function() {
+  	var passed, flag;  	
+
+  	runs(function(){
+  		passed = false;
+  		flag = false;
+  		b.test.testEnsureSuccess(validProgram, {}, function(result){
+  			passed = result;
+  			flag = true;
+  		})
+  	});
+
+  	waitsFor(function(){
+  		return flag;
+  	}, 'failure',5000);
+
+  	runs(function(){
+  		expect(passed).toBe(true);
+  	});
+  });
+
+  xit('passes on a valid (long) program', function() {
+    var passed, flag;   
+
+    runs(function(){
+      passed = false;
+      flag = false;
+      b.test.testEnsureSuccess(longProgram, {}, function(result){
+        passed = result;
+        flag = true;
+      })
+    });
+
+    waitsFor(function(){
+      return flag;
+    }, 'failure',5000);
+
+    runs(function(){
+      expect(passed).toBe(true);
+    });
+  });
+
+  it('fails on an invalid program', function() {
+  	var passed, flag;  	
+
+  	runs(function(){
+  		passed = false;
+  		flag = false;
+  		b.test.testEnsureSuccess(invalidProgram,{},function(result){
+  			passed = result;
+  			flag = true;
+  		})
+  	});
+
+  	waitsFor(function(){
+  		return flag;
+  	}, 'failure',5000);
+
+  	runs(function(){
+  		expect(passed).toBe(false);
+  	});
+  });
+
+  it('fails on a nonparsable program', function() {
+  	var passed, flag;  	
+
+  	runs(function(){
+  		passed = false;
+  		flag = false;
+  		b.test.testEnsureSuccess(nonParsableProgram,{},function(result){
+  			passed = result;
+  			flag = true;
+  		})
+  	});
+
+  	waitsFor(function(){
+  		return flag;
+  	}, 'failure',5000);
+
+  	runs(function(){
+  		expect(passed).toBe(false);
+  	});
+  });
+});
+
+describe('parsePyret', function(){
+  it('passes on a valid program', function() {
+    debugger;
+  	var passed, flag;  	
+
+  	runs(function(){
+  		passed = false;
+  		flag = false;
+  		b.test.testDeferredFunction(validProgram, {}, 'parsePyret',function(result){
+  			passed = result;
+  			flag = true;
+  		});
+  	});
+
+  	waitsFor(function(){
+  		return flag;
+  	}, 'failure',5000);
+
+  	runs(function(){
+  		expect(passed).toBe(true);
+  	});
+  });
+
+  xit('passes on a valid (long) program', function() {
```

Comment:
These were skipped because the 'long' part was making them take a very long time, as one might imagine. I will go over them again and make a decision.